### PR TITLE
core pool vtoken documentation

### PR DESCRIPTION
**Pull request by defistar**

### CODE REVIEW

File: core-pool/vtokens.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
+# vtoken
```
Comment (coreyar) [inline core-pool/vtokens.md]:
```suggestion
# vToken
```

File: core-pool/vtokens.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+## Introduction
+
+vTokens are the primary means of interacting with the Venus Protocol; when a user mints, redeems, borrows, repays a borrow, liquidates a borrow, or transfers vTokens, she will do so using the vToken contract.
```
Comment (coreyar) [inline core-pool/vtokens.md]:
```suggestion
vTokens are the primary means of interacting with the Venus Protocol. When a user mints, redeems, borrows, repays a borrow, liquidates a borrow, or transfers vTokens, they will do so using the vToken contract.
```

File: core-pool/vtokens.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+There are currently two types of vTokens: 
+1. VBep20
```
Comment (coreyar) [inline core-pool/vtokens.md]: We should use consistent capitalization for the `v` in front of names.

File: core-pool/vtokens.md [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+2. VBnb
+
+- Both types expose the EIP-20 interface
+- VBep20 wraps an underlying BEP-20 asset
+- VBnb simply wraps BNB itself
+
+The implementation of [VBep20](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBep20.sol) inherits the [VToken](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VToken.sol) contract as a base. 
+
+The implementation of [VBnb](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBNB.sol) inherits the [VToken](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VToken.sol) contract as a base.
```
Comment (coreyar) [inline core-pool/vtokens.md]:
```suggestion
Both [VBep20](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBep20.sol) and [VBnb](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBNB.sol) inherit the [VToken](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VToken.sol) contract. 
```

File: core-pool/vtokens.md [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+- Both types expose the EIP-20 interface
+- VBep20 wraps an underlying BEP-20 asset
+- VBnb simply wraps BNB itself
+
+The implementation of [VBep20](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBep20.sol) inherits the [VToken](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VToken.sol) contract as a base. 
+
+The implementation of [VBnb](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VBNB.sol) inherits the [VToken](https://github.com/VenusProtocol/venus-protocol/blob/master/contracts/VToken.sol) contract as a base. 
+
+
+Core functions which involve transferring an asset into the protocol have slightly different interfaces depending on the type which are listed below.
```
Comment (coreyar) [inline core-pool/vtokens.md]:
```suggestion
```

File: core-pool/vtokens.md [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+
+
+## Details
+
+- VToken actions:
```
Comment (coreyar) [inline core-pool/vtokens.md]: Somethings we can mention under details
 - methods that return and error
 - interest accrual happens in a few interesting places
 - initialization requirements such as comptroller, interest rate model, ect
 - events emmited
 - mention liquidations and the role VToken places with a link to the liquidator docs
 - some functions on the VToken are not included like handling reserves
 - methods like `_setInterestRateModelFresh` that can only be called by an admin can be highlighted with a callout hint

File: core-pool/vtokens.md (hunk 2 above)
Comment (defistar) [reply core-pool/vtokens.md]: applied changes

File: core-pool/vtokens.md (hunk 3 above)
Comment (defistar) [reply core-pool/vtokens.md]: updated with consistent naming standard for vToken

File: core-pool/vtokens.md (hunk 4 above)
Comment (defistar) [reply core-pool/vtokens.md]: updated

File: core-pool/vtokens.md (hunk 5 above)
Comment (defistar) [reply core-pool/vtokens.md]: updated

File: core-pool/vtokens.md (hunk 6 above)
Comment (defistar) [reply core-pool/vtokens.md]: @coreyar updating with the details as suggested above

Comment (defistar) [reply core-pool/vtokens.md]: @coreyar added event Data emitted, interest accrual instances
adding initialization requirements.

Comment (defistar) [reply core-pool/vtokens.md]: added

- methods that return and error
- interest accrual happens in a few interesting places
- initialization requirements such as comptroller, interest rate model, ect
- events emmited
- some functions on the VToken are not included like handling reserves

in progress:
- mention liquidations and the role VToken places with a link to the liquidator docs
- methods like _setInterestRateModelFresh that can only be called by an admin can be highlighted with a callout hint

@coreyar

Comment (defistar) [reply core-pool/vtokens.md]: @coreyar 

<p class="callout info">can only be called by an admin</p>

shall i set this way for all functions which are restricted to admin ?
placing them next or above function name?

Comment (defistar) [reply core-pool/vtokens.md]:
```
<p class="callout info">can only called by admin</p>
```

**Pull request closed without merging**